cmake/Helpers.cmake

```diff
@@ -0,0 +1,790 @@
+        SET_PATH( PLUGIN_DIR ${LIB_DIR}/plugins )
+        SET_PATH( DATA_DIR ${CMAKE_INSTALL_PREFIX}/share/ )
+        SET_PATH( MAN_DIR ${CMAKE_INSTALL_PREFIX}/man/man6 )
+        SET_PATH( BIN_DIR ${CMAKE_INSTALL_PREFIX}/bin )
+        SET_PATH( SYSCONF_DIR ${CMAKE_INSTALL_PREFIX}/etc )
+        SET_PATH( DOC_DIR ${CMAKE_INSTALL_PREFIX}/share/doc/gemrb )
+        SET_PATH( ICON_DIR ${CMAKE_INSTALL_PREFIX}/share/pixmaps )
+        SET_PATH( SVG_DIR ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps )
+        SET_PATH( MENU_DIR ${CMAKE_INSTALL_PREFIX}/share/applications )
+        SET_PATH( EXAMPLE_CONF_DIR ${SYSCONF_DIR} )
+        SET_PATH( METAINFO_DIR ${CMAKE_INSTALL_PREFIX}/share/metainfo )
+    ELSE (${LAYOUT} MATCHES "bundle") # Mac or iOS
+        SET(CMAKE_INSTALL_RPATH @loader_path/../Frameworks)
+        SET(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
+        SET_PATH( BIN_DIR /Applications )
+        SET_PATH( LIB_DIR @loader_path/../Frameworks )
+        SET_PATH( METAINFO_DIR "" )
+    ENDIF (${LAYOUT} MATCHES "home")
+ENDFUNCTION()
+
+FUNCTION(READ_GEMRB_VERSION)
+    # try to extract the version from the source
+    FILE(READ ${CMAKE_CURRENT_SOURCE_DIR}/gemrb/core/InterfaceConfig.h GLOBALS)
+    SET(GEMRB_VERSION "")
+    STRING(REGEX MATCH "define VERSION_GEMRB .([^\"]*)" GEMRB_VERSION "${GLOBALS}")
+    STRING(REGEX REPLACE "define VERSION_GEMRB .([^\"]*)$" "\\1" GEMRB_VERSION "${GEMRB_VERSION}")
+    IF(GEMRB_VERSION STREQUAL "") # lookup failed
+        set(GEMRB_VERSION "unknown")
+    ENDIF()
+    message(STATUS "Detected version: ${GEMRB_VERSION}")
+    SET(GEMRB_VERSION ${GEMRB_VERSION} CACHE INTERNAL "Version of GemRB")
+    unset(GLOBALS)
+ENDFUNCTION()
+
+FUNCTION(CACHE_BUILD_TYPE)
+    # If the user specifies -DCMAKE_BUILD_TYPE on the command line, take their definition
+    # and dump it in the cache along with proper documentation, otherwise set CMAKE_BUILD_TYPE
+    # to Release prior to calling PROJECT()
+    IF(DEFINED CMAKE_BUILD_TYPE)
+        SET(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE} CACHE STRING "Choose the type of build, options are: None(CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) Debug Release RelWithDebInfo MinSizeRel.")
+    ELSE(DEFINED CMAKE_BUILD_TYPE)
+        SET(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Choose the type of build, options are: None(CMAKE_CXX_FLAGS or CMAKE_C_FLAGS used) Debug Release RelWithDebInfo MinSizeRel.")
+    ENDIF(DEFINED CMAKE_BUILD_TYPE)
+ENDFUNCTION()
+
+FUNCTION(ENABLE_LINK_TIME_OPTIMIZATIONS)
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT supported)
+
+    IF(supported)
+        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE CACHE BOOL)
+    ENDIF()
+ENDFUNCTION()
+
+FUNCTION(SET_WERROR_POLICY)
+    SET(RC "")
+    STRING(REGEX MATCH "-git$" RC ${GEMRB_VERSION})
+    IF(RC STREQUAL "")
+        set (ENABLE_WERROR 0)
+    ELSEIF(NOT DISABLE_WERROR)
+        set (ENABLE_WERROR 1)
+    ENDIF()
+    MESSAGE(STATUS "ENABLE_WERROR: ${ENABLE_WERROR}")
```

Comment:
This will be deceptive, as users are passing DISABLE_WERROR that you now ignore.

Reply:
I meant the message, but misread. However the combination of git version and DISABLE_WERROR leaves the variable undefined, I guess an empty string. It would be better if it was consistent.

Reply:
The logic seems to be correct now. What name for this variable do you suggest then?